Controller/AbstractFOSRestController.php

```diff
@@ -36,7 +36,7 @@ protected function getViewHandler()
     /**
      * {@inheritdoc}
      */
-    public static function getSubscribedServices()
+    public static function getSubscribedServices(): array
```

Comment:
So this will trigger a fatal error for anyone that extends the AbstractFOSRestController and overrides this method without specifying the return type, right?

Reply:
I'm not sure how much worth is it, but I've published a merge request to your branch with a backward compatible version of this in https://github.com/mbabker/FOSRestBundle/pull/1

Reply:
> So this will trigger a fatal error for anyone that extends the AbstractFOSRestController and overrides this method without specifying the return type, right?

Right, once the return type is declared somewhere in the hierarchy all sub-classes have to have it.  So, your PR would handle that in a B/C way until 4.0 ships.  And given how service subscribers are somewhat encouraged and that it's more likely for folks to have an extended version of this method in controllers, I'd say your PR is worth the extra effort.

Reply:
@mbabker feel free to merge my pull request. after that i think that is if the build is green for this we can merge it!

Reply:
We'll need #2341 first then we should be good.